### PR TITLE
Set up analytics collection in the Butterfly app and API.

### PR DESCRIPTION
**Pull request by vingkan**

### CODE REVIEW

File: backend/analytics/views/collect.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+def post_analytics_data(request, event: AnalyticsEvent):
+    # Add timestamp to event data
+    record = event.dict()
+    record["timestamp"] = SERVER_TIMESTAMP
+
+    # Save event to database
+    db = get_db()
+    ref = db.reference("analytics")
+    # TODO: Remove this check to actually saved the event
+    if SHOULD_SAVE:
```
Comment (vingkan) [inline backend/analytics/views/collect.py:27]: Lauren will remove this check, as well as line 7, to start saving analytics events. We can also print the values to look at what they contain.

File: backend/analytics/views/collect_test.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            "page_id": "edit_profile",
+            "duration_ms": 2103,
+        },
+        "timestamp": SERVER_TIMESTAMP,
+    }
+    expected = format_json(status=200, message="Event saved successfully.")
+    assert_response_match(actual, expected)
+    # TODO: Update this test to check that the mock push is called correctly
+    # mock_push.assert_called_once_with(expected_event)
+    mock_push.assert_not_called()
```
Comment (vingkan) [inline backend/analytics/views/collect_test.py:40]: Lauren will also update this test.

File: frontend/src/app/chat/ChatHooks.js [hunk 3]
```diff
@@ -132,4 +133,9 @@ export async function sendMessage(chatId, data) {
         release: data?.release,
         chat: data?.chat,
     })
+    saveEvent('send_chat_message', {
+        community: data?.community,
+        release: data?.release,
+        chat: data?.chat,
+    })
```
Comment (vingkan) [inline frontend/src/app/chat/ChatHooks.js:140]: Lauren will add more of these `saveEvent()` calls throughout the frontend to collect more analytics information.

**Review by vingkan — COMMENTED** [review]
FYI @arushirai1 here is the change to enable saving analytics events in the frontend and backend. I left three comments showing where Lauren will make changes during our pairing session tomorrow.

**Review by arushirai1 — APPROVED** [review]
Looks good to me!

**Pull request merged**